src/Motions/PairMatch.ts

```diff
@@ -0,0 +1,90 @@
+                                       "<"  : ">",
+                                       ">"  : ">",
+                                       "("  : ")",
+                                       ")"  : ")",
+                                       "["  : "]",
+                                       "]"  : "]",
+                                       "{"  : "}", 
+                                       "}"  : "}",
+                                      };
+                                 
+    private static mapToOpeningChar = {"'"  : "'",
+                                       "\"" : "\"", 
+                                       ">"  : "<",
+                                       "<"  : "<",
+                                       ")"  : "(",
+                                       "("  : "(",
+                                       "]"  : "[",
+                                       "["  : "[",
+                                       "}"  : "{",
+                                       "{"  : "{"
+                                      };
+   
+    /** Go to the first opening char (exclusive) corresponding to 'args.character'  */
+    static matchOpening(args: {character: string}): MotionPairMatch {        
+        const obj = new MotionPairMatch();
+        obj.character = args.character;
+        obj.direction = MotionPairMatchDirection.OPENING;
+        return obj;
+    }
+    
+    /** Go to the first closing char (exclusive) corresponding to 'args.character'  */
+    static matchEnding(args: {character: string}): MotionPairMatch {        
+        const obj = new MotionPairMatch();
+        obj.character = args.character;
+        obj.direction = MotionPairMatchDirection.CLOSING;    
+        return obj;
+    }
+
+    apply(from: Position, option: {isInclusive?: boolean} = {}): Position {
+        from = super.apply(from);
+        const activeTextEditor = window.activeTextEditor;
+
+        if (! activeTextEditor || this.direction === undefined || ! this.character) {
+            return from;
+        }
+
+        const document = activeTextEditor.document;
+        let toLine = from.line;
+        let toCharacter = from.character;
+        let targetText:string = document.lineAt(toLine).text;
+
+        if (this.direction === MotionPairMatchDirection.OPENING) {
+             targetText = targetText
+                .substr(0, toCharacter)
+                .split('').reverse().join('');
+
+            var openingChar = MotionPairMatch.mapToOpeningChar[this.character];
+            const offset = targetText.indexOf(openingChar);
+
+            if (!!~offset) {
+                toCharacter -= offset;
+            }
+        }
+
+        else if (this.direction === MotionPairMatchDirection.CLOSING) {
+            targetText = targetText.substr(toCharacter + 1);
+            var closingChar = MotionPairMatch.mapToClosingChar[this.character];
+            const offset = targetText.indexOf(closingChar);
```

Comment:
Yes you are absolutely right. But how to cancel next actions if the motion is wrong in your API ? Because managing a stack of open and closing can be added easily, but I need to cancel the motion and actions following it if the user press a wrong sequence, like"cih"